### PR TITLE
Add healthcheck configuration and RBAC in route agent and register crds

### PR DESCRIPTION
**Pull request by aswinsuryan**
Add the healthcheker configuration in route agent and give
permission for routegent permission for crud operation of
routeagent resources.

Register the routeagent crds

### CODE REVIEW

File: config/rbac/submariner-route-agent/role.yaml [hunk 1]
```diff
@@ -25,3 +25,12 @@ rules:
       - create
       - update
       - delete
+  - apiGroups:
+      - submariner.io
+    resources:
+      - routeagent
+    verbs:
+      - list
```
Comment (tpantelis) [inline config/rbac/submariner-route-agent/role.yaml:34]: We'll need "get" as well.

Comment (aswinsuryan) [reply config/rbac/submariner-route-agent/role.yaml:34]: Done.

File: config/rbac/submariner-route-agent/role.yaml [hunk 2]
```diff
@@ -25,3 +25,13 @@ rules:
       - create
       - update
       - delete
+  - apiGroups:
+      - submariner.io
+    resources:
+      - routeagent
```
Comment (tpantelis) [inline config/rbac/submariner-route-agent/role.yaml]:
```suggestion
      - routeagents
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Comment by tpantelis** [comment]
I assume we'll be checking `RouteAgent` CRs in `subctl diagnose` so we'll need RBAC permissions [here](https://github.com/submariner-io/submariner-operator/blob/744f41dc3110782089c8cbfd34d1c228973b217c/config/rbac/submariner-diagnose/role.yaml#L35) we well.

**Comment by aswinsuryan** [comment]
> I assume we'll be checking `RouteAgent` CRs in `subctl diagnose` so we'll need RBAC permissions [here](https://github.com/submariner-io/submariner-operator/blob/744f41dc3110782089c8cbfd34d1c228973b217c/config/rbac/submariner-diagnose/role.yaml#L35) we well.

Done.

File: config/rbac/submariner-route-agent/role.yaml (hunk 2 above)
Comment (aswinsuryan) [reply config/rbac/submariner-route-agent/role.yaml]: Done

**Review by tpantelis — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Pull request merged**